spirv_hlsl.cpp

```diff
@@ -1196,6 +1357,52 @@ void CompilerHLSL::emit_resources()
 		emitted = true;
 	}
 
+	// HLSL requires ray payloads to be structs
+	if (execution.model == ExecutionModelClosestHitKHR || execution.model == ExecutionModelAnyHitKHR ||
+	    execution.model == ExecutionModelMissKHR)
+	{
+		auto *payload_var = get_ray_tracing_payload();
```

Comment:
This seems a bit weird. There can be multiple output payloads in GLSL, each with their own location qualifier.

Reply:
get_ray_tracing_payload takes the incoming ray payload, I made this more obvious now